src/algebra/algebra/restrict_scalars.lean

```diff
@@ -96,17 +98,34 @@ The preferred way of setting this up is `[module R M] [module S M] [is_scalar_to
 instance : module R (restrict_scalars R S M) :=
 module.comp_hom M (algebra_map R S)
 
+instance : is_scalar_tower R S (restrict_scalars R S M) :=
+⟨λ r S M, by { rw [algebra.smul_def, mul_smul], refl }⟩
+
+/--
+The `R`-algebra homomorphism from the original coefficient algebra `S` to endomorphisms
+of `restrict_scalars R S M`.
+-/
+def restrict_scalars.lsmul : S →ₐ[R] (module.End R (restrict_scalars R S M)) :=
```

Comment:
Is `restrict_scalars.module_orig` used in the type here, or only the implementation? If the latter, can you move the `local instance` to a `haveI` to make that clear?

Reply:
Good suggestion, thanks.